### PR TITLE
New feature text for CCO in 4.6

### PR DESCRIPTION
**Pull request by jeana-redhat**
For 4.6 GA Release Notes

### CODE REVIEW

File: release_notes/ocp-4-6-release-notes.adoc [hunk 1]
```diff
@@ -196,6 +196,18 @@ By default, the following network connections are checked:
...
+=== Operators
+
+[id="ocp-4-6-cloud-credential-operator-mode-support"]
+==== Support for Cloud Credential Operator modes
+
+The xref:../operators/operator-reference.adoc#cloud-credential-operator_red-hat-operators[Cloud Credential Operator (CCO)]
+can now be configured to operate in the following modes: `Mint`, `Passthrough`,
+and `Manual`. This feature provides transparency and flexibility in how the CCO
+uses cloud credentials to process `CredentialRequests` in the cluster for
+installation and other tasks.
```
Comment (jeana-redhat) [inline release_notes/ocp-4-6-release-notes.adoc:209]: New feature summary for [CO-967: CCO: explicit mode support](https://issues.redhat.com/browse/CO-967)
(Note that updated CCO content from that link is in a different branch)

File: release_notes/ocp-4-6-release-notes.adoc [hunk 2]
```diff
@@ -495,6 +507,38 @@ $ operator-sdk new <operator_name> --type=helm \
...
+credentialsMode: Mint <1>
+compute:
+- architecture: amd64
+  hyperthreading: Enabled
+...
+----
+<1> This line is added to set the `credentialsMode` parameter to `Mint`.
++
+When bypassing this check, ensure that the credentials you provide have the
+permissions that are required for the specified mode.
```
Comment (jeana-redhat) [inline release_notes/ocp-4-6-release-notes.adoc:538]: Known issue workaround for [CO-939: AWS: Allow customers to skip all credential permissions checking](https://issues.redhat.com/browse/CO-939)
(Note that updated CCO content from that link is in a different branch)

File: release_notes/ocp-4-6-release-notes.adoc [hunk 3]
```diff
@@ -495,6 +507,38 @@ $ operator-sdk new <operator_name> --type=helm \
...
++
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1829101[*BZ#1829101*])
```
Comment (jeana-redhat) [inline release_notes/ocp-4-6-release-notes.adoc:540]: This is the BZ used in 4.5.8 - is there a different reference for 4.6?

**Comment by jeana-redhat** [comment]
@joelddiaz @dgoodwin PTAL - this PR will eventually have all the blurbs for CCO stuff in 4.6, but I figured it might be better to work on this as specific items are ready vs. dropping it all in your lap later. Tagged each section in the code with a comment for reference.

File: release_notes/ocp-4-6-release-notes.adoc (hunk 1 above)
Comment (jeana-redhat) [reply release_notes/ocp-4-6-release-notes.adoc:209]: Preview link: https://rns-4_6-ga-cco-features--ocpdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-6-release-notes.html#ocp-4-6-cloud-credential-operator-mode-support

File: release_notes/ocp-4-6-release-notes.adoc (hunk 2 above)
Comment (jeana-redhat) [reply release_notes/ocp-4-6-release-notes.adoc:538]: https://rns-4_6-ga-cco-features--ocpdocs.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-6-release-notes.html#ocp-4-6-known-issues

File: release_notes/ocp-4-6-release-notes.adoc [hunk 4]
```diff
@@ -196,6 +196,18 @@ By default, the following network connections are checked:
...
 ** the external API load balancer
 
+[id="ocp-4-6-operators"]
+=== Operators
+
+[id="ocp-4-6-cloud-credential-operator-mode-support"]
+==== Support for Cloud Credential Operator modes
+
+The xref:../operators/operator-reference.adoc#cloud-credential-operator_red-hat-operators[Cloud Credential Operator (CCO)]
+can now be configured to operate in the following modes: `Mint`, `Passthrough`,
```
Comment (joelddiaz) [inline release_notes/ocp-4-6-release-notes.adoc:206]: I would suggest adding some kind of mention that the original default behavior (credentialsMode: "") is still there and is still the default mode. Something along these lines:
```suggestion
can now be explicitly configured to operate in the following modes (in addition to the previous and still default mode of operation): `Mint`, `Passthrough`,
```

File: release_notes/ocp-4-6-release-notes.adoc (hunk 2 above)
Comment (joelddiaz) [inline release_notes/ocp-4-6-release-notes.adoc:538]: probably should specify exactly what the permissions CCO needs for Mint and Passthrough

Comment (jeana-redhat) [reply release_notes/ocp-4-6-release-notes.adoc:538]: Meaning the same level of info we had in the 4.5.8 RNs? ([Mint and Passthrough bullets here](https://docs.openshift.com/container-platform/4.5/release_notes/ocp-4-5-release-notes.html#ocp-4-5-8-added-credentialsmode-parameter-for-aws-install-workaround))

If so, that info is going into the CCO entry [here](https://docs.openshift.com/container-platform/4.5/operators/operator-reference.html#cloud-credential-operator_red-hat-operators) (link is to 4.5 version, not expanded as it is for 4.6), so I think maybe linking to that makes sense to encourage reading more about it.

Let me know if you meant something else by "exactly what the permissions CCO needs".

Comment (joelddiaz) [reply release_notes/ocp-4-6-release-notes.adoc:538]: I mean that somewhere we should be able to say something like:

For CCO to operate in mint mode (for AWS), it needs these permissions: https://github.com/openshift/cloud-credential-operator/blob/master/pkg/aws/utils.go#L21
For CCO to operate in passthrough mode (for AWS), it needs the combination of all requested permissions for all CredentialsRequests for the version of OpenShift being installed/run (which can be gathered by following the steps for Manual mode).

Comment (jeana-redhat) [reply release_notes/ocp-4-6-release-notes.adoc:538]: Thank you! Yeah that will def have to be linked, but I'll make sure to explain it and link out to that info in the main docset.

**Comment by jeana-redhat** [comment]
This actually ended up covering everything I had for 4.6 - the AWS and Azure gov't support is covered by the writer who worked on installer content. So @joelddiaz lmk if my edits sufficiently address your earlier feedback and if so we can move this along :)

**Review by joelddiaz — COMMENTED** [review]
the CCO blurbs look fine to me

**Comment by jeana-redhat** [comment]
@lwan-wanglin This is ready for QA review, PTAL. There are two parts:

1. New feature summary for [CO-967: CCO: explicit mode support](https://issues.redhat.com/browse/CO-967) (lines 485-489)
2. Known issue workaround for [CO-939: AWS: Allow customers to skip all credential permissions checking](https://issues.redhat.com/browse/CO-939) (lines 877-907)

**Comment by lwan-wanglin** [comment]
@jeana-redhat, The content looks good to me! Thanks.

**Comment by jeana-redhat** [comment]
oops forgot to tag @openshift/team-documentation and apply peer review label

**Comment by bmcelvee** [comment]
LGTM!

**Pull request merged**